android/src/main/java/bbct/android/common/activity/MainActivity.java

```diff
@@ -167,17 +167,29 @@ public boolean onCreateOptionsMenu(Menu menu) {
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
-        int itemId = item.getItemId();
-
-        if (itemId == R.id.about_menu) {
-            this.getSupportFragmentManager()
-                    .beginTransaction()
-                    .replace(R.id.fragment_holder, new About(), FragmentTags.ABOUT)
-                    .addToBackStack(FragmentTags.ABOUT)
-                    .commit();
-            return true;
+        int menuId = item.getItemId();
+
+        switch (menuId) {
+            case R.id.about_menu:
+                this.getSupportFragmentManager()
+                        .beginTransaction()
+                        .replace(R.id.fragment_holder, new About(), FragmentTags.ABOUT)
+                        .addToBackStack(FragmentTags.ABOUT)
+                        .commit();
+                return true;
+            case android.R.id.home:
+                this.onBackPressed();
+                return true;
         }
 
         return super.onOptionsItemSelected(item);
     }
+
+    @Override
+    public void onBackStackChanged() {
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(getSupportFragmentManager().getBackStackEntryCount() > 0);
```

Comment:
When `BaseballCardList` is visible will the backstack always be empty? Why isn't `BaseballCardList` itself added to the backstack? (See previous comment when creating `BaseballCardList`.)

I'm not saying that this is incorrect. I'm asking for the reasoning behind this implementation.

Reply:
This is just to disable the actionBar.setDisplayHomeAsUpEnabled Toolbar icon if we only have one item left at the backstack.

Reply:
@codeguru42 I think I messed up on this one, I've made a few changes now on the MainActivity, yes it should be > 1, as I am supposed to add the CARD_LIST to backstack.